docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -45,11 +46,18 @@ with the following additions:
    must be omitted
 4. `repeated` fields of scalar numeric types must use
    [packed encoding](https://developers.google.com/protocol-buffers/docs/encoding#packed)
-   by default.
-5. Variant encoding of integers must not be longer than needed.
+5. Variant encoding must not be longer than needed:
+    1. The highest 6 bits for 10-byte varints must be `0`
```

Comment:
I'm not sure I understand this rule. `2^63` has varint encoding of `80 80 80 80 80 80 80 80 80 01` (10 bytes), the highest 6 bits are not 0.

Reply:
Implementation: https://github.com/lazyledger/protobuf3-solidity-lib/blob/v0.4.2/contracts/ProtobufLib.sol#L110-L117

`FF FF FF FF FF FF FF FF FF 7F` is a 10-byte varint whose highest 6 bits are `1` (recall that 10-byte varints are 10 groups of 7 bits, i.e. 70 bits of data, and 70-6 = 64).

Test showing this: https://github.com/lazyledger/protobuf3-solidity-lib/blob/v0.4.2/test/TestFixture.js#L18-L25

Reply:
@adlerjohn  - can you add / link some reference here? When reading this document, I have no idea were `6 bits` and `10 byte` comes from, and why we want to enforce that rule.

Reply:
Sure

Reply:
Clarified in 6a6dbbb.